tests/test_ai/test_priority_algorithm/test_mutations.py

```diff
@@ -0,0 +1,263 @@
+from ai.priority_algorithm.priority_teamset import PriorityTeamSet, PriorityTeam
+from models.student import Student
+from models.team import Team
+
+
+class TestPriority(Priority):
+    """
+    A custom priority for this test.
+    It wants students 1 and 2 to be in the same team
+    """
+
+    def validate(self):
+        return True
+
+    def satisfied_by(self, students: List[Student]) -> bool:
+        ids = [student.id for student in students]
+        return 1 in ids and 2 in ids
+
+
+def print_priority_team_set(priority_team_set: PriorityTeamSet):
+    """
+    For debugging purposes
+    """
+    for priority_team in priority_team_set.priority_teams:
+        print(f"Team {priority_team.team.id}: {priority_team.student_ids}", end=" ")
+    print()
+
+
+def equal_priority_team_sets(a: PriorityTeamSet, b: PriorityTeamSet) -> bool:
+    """
+    Checks if two priority team sets are equal
+    """
+    if len(a.priority_teams) != len(b.priority_teams):
+        return False
+    for priority_team_a, priority_team_b in zip(a.priority_teams, b.priority_teams):
+        if priority_team_a.team.id != priority_team_b.team.id:
+            return False
+        if priority_team_a.student_ids != priority_team_b.student_ids:
+            return False
+    return True
+
+
+def create_new_priority_team_set(
+    num_teams: int, num_students: int
+) -> (PriorityTeamSet, Dict[int, Student]):
+    """
+    Creates a PriorityTeamSet and a student dict
+    """
+    team_list: List[Team] = [Team(team_id) for team_id in range(num_teams)]
+    for student_id in range(num_students):
+        student: Student = Student(student_id)
+        team_list[student_id % num_teams].add_student(student)
+    priority_team_list: List[PriorityTeam] = [
+        PriorityTeam(team, [student.id for student in team.students])
+        for team in team_list
+    ]
+    priority_team_set: PriorityTeamSet = PriorityTeamSet(priority_team_list)
+    student_dict: Dict[int, Student] = {}
+    for team in team_list:
+        for student in team.students:
+            student_dict[student.id] = student
+    return priority_team_set, student_dict
+
+
+def get_priority_team(team_id: int, priority_team_set: PriorityTeamSet) -> PriorityTeam:
+    """
+    Gets the priority team with the given team id
+    """
+    if team_id < 0:
+        raise ValueError(f"Team id must be positive, not {team_id}")
+    if team_id >= len(priority_team_set.priority_teams):
+        raise ValueError(
+            f"Team id must be less than {len(priority_team_set.priority_teams)}, not {team_id}"
+        )
+    if priority_team_set.priority_teams[team_id].team.id == team_id:
+        return priority_team_set.priority_teams[team_id]
+    for priority_team in priority_team_set.priority_teams:
+        if priority_team.team.id == team_id:
+            return priority_team
+    raise ValueError(f"Team {team_id} not found")
+
+
+class TestMutations(unittest.TestCase):
```

Comment:
tbh I would give testing each mutation it's own class for testing if it's going to be this juicy